database/test/db-admin-personaroles.test.js

```diff
@@ -21,4 +21,100 @@ describe('Admin persona roles view', () => {
       assert.equal(personaRoles.length, 0)
     })
   })
+
+  describe('Test Insert personas roles', () => {
+    let testPersonaKey
+    let validRoleKey
+    const invalidPersonaKey = 0
+    const invalidRoleKey = 0
+    before(async () => {
+      const { key: roleKey } = await usherDb('roles').select('key').first()
+      validRoleKey = roleKey
+      const { key: tenantkey } = await usherDb('tenants').select('key').first()
+      const [persona] = await usherDb('personas').insert({ tenantkey, sub_claim: 'personarole@test' }).returning('key')
+      testPersonaKey = persona.key
+    })
+
+    it('Should return an array of inserted personaroles records', async () => {
+      const personaRoles = await adminPersonaRoles.insertPersonaRoles(testPersonaKey, [validRoleKey])
+      assert.equal(personaRoles.length, 1)
+      assert.equal(personaRoles[0].personakey, testPersonaKey)
+      assert.equal(personaRoles[0].rolekey, validRoleKey)
+    })
+
+    it('Should fail due to invalid persona key', async () => {
+      try {
+        await adminPersonaRoles.insertPersonaRoles(invalidPersonaKey, [validRoleKey])
+      } catch (err) {
+        assert.equal(!!err, true)
+      }
+    })
+
+    it('Should fail due to invalid role key', async () => {
+      try {
+        await adminPersonaRoles.insertPersonaRoles(testPersonaKey, [invalidRoleKey])
+      } catch (err) {
+        assert.equal(!!err, true)
+      }
+    })
+
+    it('Should fail due to duplicate role', async () => {
```

Comment:
@k1ch  should this fail, or should the API / database do a unique and insert the "one" role to the persona?

Reply:
This specific test is on the DB layer, and I believe the appropriate behavior of the function should be to throw an error due to duplication.
Also, regarding API behavior, imagine the body request is `[ 5 ]`, and the user already has this role. If that is ignored, we will respond with a `201` status code, which I can argue is a confusing behavior. Having said that, it is more about our preference.

Reply:
@k1ch  I can see your point at the db level. 

With regards to the API, I agree that a 201 response code for a `POST` verb is confusing since POST should be used for "creating" something. 

I wonder in this case if we should change the verb to `PUT`, which should be idempotent. Te response can then be either a 200 or 204. 

Same with removing a role (or permission) from a persona, that should probably use the `DELETE` verb and be idempotent. 

What do you think on making this change? I think this same logic still holds even with the API taking an array of role IDs. 

[MDN Docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) for reference.

Reply:
I agree Michael, if we prefer to avoid having `409` errors and make API usage easier, we should go with `PUT` method. In that case I need to also update `POST /persona/{key}/permissions` to stay consistent.